pkg/config/config.go

```diff
@@ -37,6 +38,28 @@ type Config struct {
 
 // Validate does static validation of the configuration
 func (c Config) Validate() error {
+	// Validate kernel version
+	// For valid versions see https://aquasecurity.github.io/tracee/latest/getting-started/installing/prerequisites/
+
+	// Default minimum kernel version
+	minimalKernelVersion := "5.4"
+	if c.OSInfo.GetOSReleaseID() == helpers.CENTOS {
+		// In centos tracee is supported in 4.18 kernel as well
+		minimalKernelVersion = "4.18"
+	}
+
+	res, err := c.OSInfo.CompareOSBaseKernelRelease(minimalKernelVersion)
+	if err != nil {
+		// TODO  what should we do?
+		return fmt.Errorf("couldn't check kernel version compatibility: %v", err)
+	}
+	if res == helpers.KernelVersionNewer {
+		return fmt.Errorf("unsupported kernel version, must be at least %s", minimalKernelVersion)
+	}
+	if res == helpers.KernelVersionInvalid {
+		return fmt.Errorf("invalid kernel version to compare: %s", minimalKernelVersion)
```

Comment:
Can this happen?

Reply:
Kernels < 5.2 (or even 5.4) that aren't RHEL/CentOS/Alma/Rocky/Oracle would probably be unsupported (unless we start testing and open issues for the load problems they likely have. 

We have to have in mind the following:

- If the kernel isn't supported by a DISTRO or by UPSTREAM (kernel.org) then we can't support.
- If the kernel does not support BTF, it has to be present in BTFHUB for us to support.

We probably work in all kernels >= 5.4 with embedded BTF I believe (despite the distro version).

Reply:
Kernels supported by upstream:

![image](https://github.com/aquasecurity/tracee/assets/7395852/c74901f7-2040-458a-b027-5b6f0ebeb1fd)